src/frontend/src/common/components/library/data_subview/components/DownloadModal/index.tsx

```diff
@@ -220,12 +221,14 @@ const DownloadModal = ({
 
   function getDownloadButton(): JSX.Element | undefined {
     // button will have different functionality depending on download type selected
+    const metadataFilename = `${downloadPrefix}_metadata.tsv`;
+
     if (isMetadataSelected && !isFastaSelected) {
```

Comment:
It seems like there are 4 separate CSV definitions here depending on various permutations of isMetadataSelected and isFastaSelected, but the only the that (seems to?) varies is the click handler. I wonder if we can determine the click handler using conditional logic before render, and then eliminate 3/4 of the CSVLink components here.

Reply:
yeah i think that could definitely be improved.. were you thinking make the csvlink it's own component and pass the click handler as props?